manifests/metadata/provider.pp

```diff
@@ -24,13 +24,24 @@
     content => template("${module_name}/shibboleth/metadata_providers/_provider.erb"),
   }
 
+  ensure_resource('file', "${shibboleth_idp::shib_install_base}/metadata",
+    {
+      ensure  => directory,
+      owner   => $owner,
+      group   => $group,
+      mode    => $mode,
+      require => File[$shibboleth_idp::shib_install_base],
+    }
+  )
```

Comment:
The issue is the concat fragment. The main concat is in the metadata class and trying to require the entire metadata class creates a dependency cycle. I think puppet will have to order the individual resources rather than the entire class.

This is the one that caught me on our DC build. It worked fine without the require on Vagrant, but the order was wrong when run via the PuppetMaster. It tried to created the SP metadata files before the metadata directory was created by the class.